benchmark/src/main/scala/fs2/benchmark/FlowInteropBenchmark.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2013 Functional Streams for Scala
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package fs2
+package benchmark
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+
+import org.openjdk.jmh.annotations.{
+  Benchmark,
+  BenchmarkMode,
+  Mode,
+  OutputTimeUnit,
+  Param,
+  Scope,
+  State
+}
+
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.Flow.{Publisher, Subscriber, Subscription}
+
+import scala.concurrent.Future
+
+@State(Scope.Thread)
+@BenchmarkMode(Array(Mode.Throughput))
+@OutputTimeUnit(TimeUnit.SECONDS)
+class FlowInteropBenchmark {
+  @Param(Array("1024", "5120", "10240"))
+  var totalElements: Int = _
+
+  @Param(Array("1000"))
+  var iterations: Int = _
+
+  @Benchmark
+  def fastPublisher(): Unit = {
+    def publisher =
+      new Publisher[Int] {
+        override final def subscribe(subscriber: Subscriber[? >: Int]): Unit =
+          subscriber.onSubscribe(
+            new Subscription {
+              @volatile var i: Int = 0
+              @volatile var canceled: Boolean = false
+
+              override final def request(n: Long): Unit = {
+                Future {
+                  var j = 0
+                  while (j < n && i < totalElements && !canceled) {
+                    subscriber.onNext(i)
+                    i += 1
+                    j += 1
+                  }
+
+                  if (i == totalElements || canceled) {
+                    subscriber.onComplete()
+                  }
+                }(global.compute)
+
+                // Discarding the Future so it runs in the background.
+                ()
+              }
+
+              override final def cancel(): Unit =
+                canceled = true
+            }
+          )
+      }
+
+    val stream =
+      interop.flow.fromPublisher[IO](publisher, chunkSize = 512)
+
+    val program =
+      stream.compile.toVector
```

Comment:
I was not sure what was the best way to collect the results.
My other two options were doing something like `compile.fold(0)(_ + _)` or a `.foreach { _ => IO.unit }.compile.drain`

Reply:
@armanbilge I guess I was worried that something like that could be optimized and thus we would not see if the production of elements was actually faster or not.
Same about publishing `Ints` over `Units`, I kind of wanted to ensure we actually published and consumed something.

But, I guess that given `Stream` could produce any side-effects, there can't be any optimization that avoid actually producing the `Chunks` even if their values are going to be discarded.
Thus, yeah probably pushing `Units` and doing `compile.drain` is the best idea.